### ISSUE TITLE
Problems with @charset in CSS

### ISSUE THREAD

**Issue by edent** [litespeedtech/lscache_wp#184]
I think I've found two small bugs.

In `private function processAtRulesAndSelectors($css)`
https://github.com/litespeedtech/lscache_wp/blob/master/litespeed-cache/lib/css_min.class.php#L701

```
// @charset handling
if (preg_match($this->charsetRegex, $css, $matches)) {
        // Keep the first @charset at-rule found
        $charset = $matches[0];
        // Delete all @charset at-rules
        $css = preg_replace($this->charsetRegex, '', $css);
}
```

But `$matches` is not defined in that function. So the first `@charset` is replaced

Secondly, in https://github.com/litespeedtech/lscache_wp/blob/master/litespeed-cache/inc/optimizer.class.php#L149

Has the regex 
```
$data = preg_replace( '/@charset[^;]+;\\s*/', '', $data ) ;
```

This is incorrect. It should end `;\s*/` the double slash causes it to fail. See https://www.phpliveregex.com/p/ujR#tab-preg-replace for a demo

Basically, I think that `@charset` should *not* be removed from the start of a file. But I think these two functions don't work as intended.

Sorry if I haven't explained this well!

**Comment by stanleybz** [comment]
Hi Edent, thanks for reporting.

The $match in preg_match is for fill in the matched string, so you can see it's used inside the condition for keeping the first @charset.

And the additional backslash is used for escape on PHP, the result is the same in this situation.

We will consider removing the whole row as you are correct that the charset should be kept on each CSS file.
